lib/dirac_improved_staggered.cpp

```diff
@@ -51,11 +51,16 @@ namespace quda {
 			      const QudaParity parity) const
   {
     checkParitySpinor(in, out);
-
     if (checkLocation(out, in) == QUDA_CUDA_FIELD_LOCATION) {
+#ifdef USE_LEGACY_DSLASH
```

Comment:
One of the improvements from the new code vs the old code is that the dslash kernels can be applied to full fields.  So you can replace the two Deo and Doe calls in `DiracStaggered::M` and `DiracImprovedStaggered::M` with a single kernel call.  See what I did in dirac_wilson.cpp as an example.

Reply:
fixed in 
https://github.com/lattice/quda/pull/775/commits/f631634ac5f15344b9d60423e0732d41551bcb26